tests/infra-validations.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+#Fail on first error
+set -e
+
+#Import functions
+. ./functions.sh
+
+TEST_WORKSPACE="$(pwd)/spk-env"
+[ ! -z "$SPK_LOCATION" ] || { echo "Provide SPK_LOCATION"; exit 1;}
+[ ! -z "$ACCESS_TOKEN_SECRET" ] || { echo "Provide ACCESS_TOKEN_SECRET"; exit 1;}
+[ ! -z "$AZDO_PROJECT" ] || { echo "Provide AZDO_PROJECT"; exit 1;}
+[ ! -z "$AZDO_ORG" ] || { echo "Provide AZDO_ORG"; exit 1;}
+[ ! -z "$ACR_NAME" ] || { echo "Provide ACR_NAME"; exit 1;}
+[ ! -z "$SP_APP_ID" ] || { echo "Provide SP_APP_ID"; exit 1;}
+[ ! -z "$SP_PASS" ] || { echo "Provide SP_PASS"; exit 1;}
+[ ! -z "$SP_TENANT" ] || { echo "Provide SP_TENANT"; exit 1;}
+AZDO_ORG_URL="${AZDO_ORG_URL:-"https://dev.azure.com/$AZDO_ORG"}"
+
+
+echo "TEST_WORKSPACE: $TEST_WORKSPACE"
+echo "SPK_LOCATION: $SPK_LOCATION"
+echo "AZDO_PROJECT: $AZDO_PROJECT"
+echo "AZDO_ORG: $AZDO_ORG"
+echo "AZDO_ORG_URL: $AZDO_ORG_URL"
+echo "ACR_NAME: $ACR_NAME"
+
+
+terraform_template_dir=discovery-tf-template
+tf_template_version=v0.0.1
+infra_hld_dir=discovery-infra-hld
+services_full_dir="$TEST_WORKSPACE/$mono_repo_dir/$services_dir"
+
+validation_test_yaml="rg_name: <insert value>"
+
+shopt -s expand_aliases
+alias spk=$SPK_LOCATION
+echo "SPK Version: $(spk --version)"
+
+echo "Running from $(pwd)"
+if [ -d "$TEST_WORKSPACE"  ]; then rm -Rf $TEST_WORKSPACE; fi
+
+if [ ! -d "$TEST_WORKSPACE" ]; then
+  echo "Directory '$TEST_WORKSPACE' does not exist"
+  mkdir $TEST_WORKSPACE
+  echo "Created '$TEST_WORKSPACE'"
+fi
+
+cd $TEST_WORKSPACE
+
+# Setup simple TF Template ------------------
+mkdir $terraform_template_dir
+cd $terraform_template_dir
+git init
+mkdir template
+cd template
+tfTemplate=$'resource "azurerm_resource_group" "example"{\n name= "${var.rg_name}"\n location = "${var.rg_location}"\n}'
+tfVars=$'variable "rg_name" { \n type = "string" \n } \n variable "rg_location" { \n type = "string" \n }'
+touch main.tf variables.tf
+echo "$tfVars" >> variables.tf
+echo "$tfTemplate" >> main.tf
+file_we_expect=("variables.tf" "main.tf")
+validate_directory "$TEST_WORKSPACE/$terraform_template_dir/template" "${file_we_expect[@]}" >> $TEST_WORKSPACE/log.txt
+# The TF Template requires a git release for a version to be targeted for spk scaffold
+git add -A
+
+# See if the remote repo exists
+repo_exists $AZDO_ORG_URL $AZDO_PROJECT $terraform_template_dir
+
+# Create the remote repo for the local repo
+created_repo_result=$(az repos create --name "$terraform_template_dir" --org $AZDO_ORG_URL --p $AZDO_PROJECT) >> $TEST_WORKSPACE/log.txt
+
+# Extract out remote repo URL from the above result
+remote_repo_url=$(echo $created_repo_result | jq '.remoteUrl' | tr -d '"' )
```

Comment:
what happens if the call above fails?

Reply:
Then the integration test will fail. This is also how `spk services` is executing their integration tests; they rely on the `az cli` to create their mono, manifests and HLD repos.